app/invocation/invocation_action_card.tsx

```diff
@@ -393,8 +394,29 @@ export default class InvocationActionCardComponent extends React.Component<Props
       .catch((e) => console.error(e));
   }
 
+  // For firecracker actions, VM metadata is stored in the auxiliary metadata field
+  // of the execution metadata. Try to decode it into an object if it exists.
+  private getFirecrackerVMMetadata(): firecracker.VMMetadata | null | undefined {
+    const auxiliaryMetadata = this.state.actionResult?.executionMetadata?.auxiliaryMetadata;
+    if (!auxiliaryMetadata) {
+      return this.getDeprecatedFirecrackerVMMetadata();
+    }
+    for (const metadata of auxiliaryMetadata) {
+      if (metadata.typeUrl.includes("firecracker.VMMetadata")) {
```

Comment:
nit: can we do an equality check on the `typeUrl` instead of `.includes()` ? or does the URL include some contents we aren't expecting?

Reply:
Changed. I thought the includes might be slightly less brittle, but don't feel strongly